src/main/java/com/atlassian/db/replica/internal/state/ConnectionState.java

```diff
@@ -184,6 +184,7 @@ private Connection prepareReadConnection(RouteDecisionBuilder decisionBuilder) t
         if (consistency.isConsistent(() -> readConnection.get(decisionBuilder))) {
             if (getState().equals(COMMITED_MAIN)) {
                 closeConnection(writeConnection, decisionBuilder);
+                parameters.initialize(readConnection.get(decisionBuilder));
```

Comment:
Removing the fix is detected by the new test 👍 
`org.postgresql.util.PSQLException: Cannot commit when autoCommit is enabled.`

Reply:
When I bring it out of the `if`s, it's green.

- [ ] move outside of ifs or catch in tests

Reply:
It makes sense to populate the parameters for every change to the replica.

Reply:
That was the last place where `State.NOT_INITIALISED` was explicitly checked. But we should preserve it, because it's still implicitly checked in `else` clauses like in `if (state.equals(REPLICA)) else { ... }`. And it's still exposed as API.

Reply:
@dagguh I see we have the same issue with the write connection. It seems it can pass without initialisation. I'm working on the repro.

Reply:
aye aye